src/rgw/rgw_op.h

```diff
@@ -197,6 +213,36 @@ class RGWGetObj_CB : public RGWGetDataCB
   }
 };
 
+class RGWGetObj_Filter : public RGWGetDataCB
+{
+protected:
+  RGWGetDataCB* next;
+public:
+  RGWGetObj_Filter(RGWGetDataCB* next): next(next) {}
+  virtual ~RGWGetObj_Filter() {}
+  /**
+   * Passes data through filter.
+   * Filter can modify content of bl.
+   * When bl_len == 0 , it means 'flush
+   */
+  virtual int handle_data(bufferlist& bl, off_t bl_ofs, off_t bl_len) override {
+    return next->handle_data(bl, bl_ofs, bl_len);
+  }
+  /**
+   * Flushes any cached data. Used by RGWGetObjFilter.
+   * Return logic same as handle_data.
+   */
+  virtual int flush() override {
+    return next->flush();
+  }
+  /**
+   * Allows filter to extend range required for successful filtering
+   */
+  virtual void fixup_range(off_t& bl_ofs, off_t& bl_len) override {
```

Comment:
the naming of `bl_len` is confusing in `RGWGetObj_Filter` and `RGWGetObj_Decompress`; it sounds like it should be `bl_end`, like it is in `RGWGetDataCB`